int64 reference_choice
---
bool success
